lib/automate/__init__.py

```diff
@@ -0,0 +1,75 @@
+from fuzzywuzzy import process as fuzzy
+from importlib import import_module
+
+from lib import Error
+from lib.settings import SETTINGS
+
+
+class Automate:
+    def __init__(self):
+        """ Register new modules here """
+        self.modules = [
+            import_module("lib.automate.modules.send").Send,
+        ]
+        self.verbs = {}
+        for module in self.modules:
+            for verb in module.verbs:
+                self.verbs[verb] = module
+
+    def get_verbs(self):
+        """
+        :return: A list of keywords for the registered automation modules.
+        :rtype: list[string]
+        """
+        return list(self.verbs.keys())
+
+    def get_senders(self):
+        """
+        :return: A list of names of users listed in the SETTINGS.
+        :rtype: list[string]
+        """
+        return list(SETTINGS["users"].keys())
+
+    def run(self, module_name, to, when, body, sender):
```

Comment:
Here, maybe we should let the module itself handle a list of parameters? Which could be any type. This looks fine for the email part and we could always make it more abstract in the future.

Reply:
I agree that a more abstract parameter list would be good for the automation side. My concern is that the NLP must know where to put what data, even when it doesn't know anything about the automation module's function. That's why I chose these, since they could be applicable on a variety of tasks, and still clearly signal what data to put where.

But if we can solve the problem of knowing where to put the data for the NLP I'm all for a more abstract parameter list.